pkg/apk/impl/fs/memfs_test.go

```diff
@@ -235,3 +235,34 @@ func TestMemFSHardlink(t *testing.T) {
 		require.Equal(t, originalContent, linkContent, "content of %s should be %s", linkName, originalContent)
 	})
 }
+
+func TestMemFSMidLevelSymlink(t *testing.T) {
```

Comment:
Yup. And it did when I put it in without the fix. Would have been nice if we had this _before_, so we wouldn't have had the issue in the first place, but this way Dan (the other) got to find something fun. 😄

Reply:
Okay cool. I might be looking at this wrong! Here's what I did locally:

```console
$ gh pr checkout 620
$ git checkout main -- pkg/apk/impl/fs/memfs.go
$ go test -v ./pkg/apk/impl/fs/ -run TestMemFSMidLevelSymlink
=== RUN   TestMemFSMidLevelSymlink
--- PASS: TestMemFSMidLevelSymlink (0.00s)
PASS
ok      chainguard.dev/apko/pkg/apk/impl/fs     0.284s
```

Reply:
im having same results

Reply:
Now I am really confused because:

1. I know I had that test failing before.
2. That is the exact case that caused the failure from this morning.

Digging.

Reply:
Try it.

Reply:
Looks fixed, thanks! 🎉